packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -26,14 +26,41 @@ typedef PointerExitEventListener = void Function(PointerExitEvent event);
 /// Used by [MouseTrackerAnnotation], [MouseRegion] and [RenderMouseRegion].
 typedef PointerHoverEventListener = void Function(PointerHoverEvent event);
 
+// A key, so that (_IdenticalKey<T>(a) == _IdenticalKey<T>(b)) is equivalent to
+// identical(a, b).
+class _IdenticalKey<T> implements Key {
+  _IdenticalKey(this.value);
+
+  final T value;
+
+  // Don't return `identityHashCode(value)` directly since the hash of this key
+  // needs to be different from that of its value.
+  @override
+  int get hashCode => identityHashCode(T) ^ identityHashCode(value);
```

Comment:
I really want to reduce the overhead of `hashCode` since it will be used a lot every frame. And since it's a private class we don't need a very secure hashing. The only feature we need is that `theKey.hashCode != theKey.value.hashCode`.

I changed it to a XOR with a const value. What do you think?